dot/parachain/candidate-validation/candidate_validation_test.go

```diff
@@ -677,3 +669,215 @@ func Test_performBasicChecks(t *testing.T) {
+			uint32(3),
+			gomock.AssignableToTypeOf(parachaintypes.OccupiedCoreAssumption{})).
+		Return(&expectedPersistedValidationDataSmallMax, nil).AnyTimes()
+	mockInstance.EXPECT().
+		ParachainHostValidationCode(uint32(3), gomock.AssignableToTypeOf(parachaintypes.OccupiedCoreAssumption{})).
+		Return(&validationCode, nil).AnyTimes()
+
+	mockInstance.EXPECT().
+		ParachainHostPersistedValidationData(
+			uint32(4),
+			gomock.AssignableToTypeOf(parachaintypes.OccupiedCoreAssumption{})).
+		Return(&expectedPersistedValidationData, nil).AnyTimes()
+	mockInstance.EXPECT().
+		ParachainHostValidationCode(uint32(4), gomock.AssignableToTypeOf(parachaintypes.OccupiedCoreAssumption{})).
+		Return(&validationCode, nil).AnyTimes()
+
+	mockInstance.EXPECT().
+		ParachainHostPersistedValidationData(
+			uint32(5),
+			gomock.AssignableToTypeOf(parachaintypes.OccupiedCoreAssumption{})).
+		Return(&expectedPersistedValidationData, nil).AnyTimes()
+	mockInstance.EXPECT().
+		ParachainHostValidationCode(uint32(5), gomock.AssignableToTypeOf(parachaintypes.OccupiedCoreAssumption{})).
+		Return(&validationCode, nil).AnyTimes()
+
+	bd, err := scale.Marshal(BlockDataInAdderParachain{
+		State: uint64(1),
+		Add:   uint64(1),
+	})
+	require.NoError(t, err)
+	pov := parachaintypes.PoV{
+		BlockData: bd,
+	}
+
+	type args struct {
+		runtimeInstance  parachainruntime.RuntimeInstance
+		candidateReceipt parachaintypes.CandidateReceipt
+		pov              parachaintypes.PoV
+	}
+	tests := map[string]struct {
+		args          args
+		want          *ValidationResult
+		expectedError error
+		isValid       bool
+	}{
+		"invalid_pov_hash": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt2,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &povHashMismatch,
+			},
+			isValid: false,
+		},
+		"invalid_pov_size": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt3,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &paramsTooLarge,
+			},
+		},
+		"code_mismatch": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt4,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &codeHashMismatch,
+			},
+			isValid: false,
+		},
+		"bad_signature": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt5,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				InvalidResult: &badSignature,
+			},
+			isValid: false,
+		},
+		"happy_path": {
+			args: args{
+				runtimeInstance:  mockInstance,
+				candidateReceipt: candidateReceipt,
+				pov:              pov,
+			},
+			want: &ValidationResult{
+				ValidResult: &ValidValidationResult{
+					CandidateCommitments: parachaintypes.CandidateCommitments{
+						UpwardMessages:     nil,
+						HorizontalMessages: nil,
+						NewValidationCode:  nil,
```

Comment:
the default values are nil here, so no point setting those

Reply:
Removed.